python/ray/node.py

```diff
@@ -240,6 +240,10 @@ def __init__(self,
 
         if not connect_only:
             self.start_ray_processes()
+            # we should update the address info after the node has been started
+            ray._private.services.wait_for_node(self.redis_address,
```

Comment:
We need to update the node info after the node has been started up. Otherwise, we could not find the given node info from the redis. It will return the head node info. The `wait_for_node` could raise `TimeoutError`.

Reply:
I think the error message here is pretty unfriendly. Can you re-catch the exception and raise better error messages? It could be something like `The current node information hasn't been updated within X seconds, this could happen because of X and Y.`

Reply:
Btw, there's one concern with this approach.

We made GlobalStateAccessor to be lazily initialized because this increase the number of connections to the head node (since it creates a new connection to the head node from every worker), and this change will make that useless. Is there any other way to see if the node info is updated without using GlobalStateAccessor?

Reply:
Hi @rkooo567, we need to call `ray._private.services.get_address_info_from_redis` both [connect only](https://github.com/ray-project/ray/blob/master/python/ray/node.py#L185) or [not](https://github.com/ray-project/ray/blob/master/python/ray/node.py#L241). And they both need to initialize the GlobalStateAccessor.